client/landing/gutenboarding/stores/onboard/actions.ts

```diff
@@ -121,7 +122,9 @@ export function* createSite(
 			site_information: {
 				title: siteTitle,
 			},
-			site_creation_flow: 'gutenboarding',
+			site_creation_flow: isEnabled( 'gutenboarding/site-editor' )
```

Comment:
Curious, why would we need to alter this?

I'd expect simple redirection be enough in https://github.com/Automattic/wp-calypso/pull/43246/files#diff-920096199c37aa596a13663b1e50eb32L127

Reply:
> Curious, why would we need to alter this?

See D44941-code -- we need to communicate to the site creation endpoint that we need to enable the Site Editor for the newly created site. (This follows the same pattern that was previously used for dotcom-FSE).

> I'd expect simple redirection be enough in https://github.com/Automattic/wp-calypso/pull/43246/files#diff-920096199c37aa596a13663b1e50eb32L127

Without Site Editor enabled, that redirect wouldn't work, and redirect back to the post editor.

Reply:
I could be convinced that changing the flow name is bad because other stuff on the backend also relies on the flow name being `gutenboarding`. We'd have to add a different parameter and intercept it in site creation to turn on the site editor

Reply:
Yeah, there's lots all around relying on that (including doing manual analytics based on this).

It's also now harder to know that we should check for both flow-names when adding new Gutenboarding features elsewhere. So if this must stay, I'd prefer having it extracted to a separate centralized helper function.

Reply:
great! I've switched the flow name back in #43349